### PR TITLE
Galera tuning parameters from reading logfiles

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Comment by dirkmueller** [comment]
(not tested)

File: chef/cookbooks/mysql/templates/default/galera.cnf.erb [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+# Disable Query Cache completely
+query_cache_size=0
+query_cache_type=0
+
 expire_logs_days = <%= @expire_logs_days %>
 
 # SST method
 wsrep_sst_method = xtrabackup-v2
 wsrep_sst_auth = <%= @sstuser %>:<%= @sstuser_password %>
+wsrep_slave_threads = 4
```
Comment (jsuchome) [inline chef/cookbooks/mysql/templates/default/galera.cnf.erb]: I've added an option for `wsrep_slave_threads` in https://github.com/crowbar/crowbar-openstack/pull/1410

**Comment by rhafer** [comment]
@dirkmueller This needs a rebase. As #1410 was merged.

**Comment by vuntz** [comment]
Is this still WIP, or can we merge?

File: chef/cookbooks/mysql/templates/default/galera.cnf.erb [hunk 2]
```diff
@@ -1,17 +1,28 @@
 [mysqld]
-wsrep_provider=/usr/lib64/galera-3/libgalera_smm.so
-wsrep_cluster_address="<%= @cluster_addresses %>"
-innodb_autoinc_lock_mode=2
-innodb_doublewrite=1
-query_cache_size=0
-wsrep_on=ON
+wsrep_on = ON
+wsrep_provider = /usr/lib64/galera-3/libgalera_smm.so
+wsrep_provider_options = "gmcast.listen_addr=tcp://<%= @node_address %>:4567"
```
Comment (vuntz) [inline chef/cookbooks/mysql/templates/default/galera.cnf.erb]: We have this option a few lines below (line 15).

**Comment by dirkmueller** [comment]
Finally found time to test it.

**Pull request merged**